maxq10xx/maxq10xx-wolfssl-client.c

```diff
@@ -0,0 +1,311 @@
+/* maxq10xx-wolfssl-client.c
+ *
+ * Based on client-tls-pkcallback.c
+ *
+ * Copyright (C) 2006-2022 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* This example shows how to write a simple TLS client that uses the features
+ * of the Analog Devices MAXQ 1065 and 1080. */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <unistd.h>
+
+/* wolfSSL */
+#ifndef WOLFSSL_USER_SETTINGS
+    #include <wolfssl/options.h>
+#endif
+#include <wolfssl/ssl.h>
+#include <wolfssl/wolfcrypt/sha256.h>
+#include <wolfssl/wolfcrypt/cryptocb.h>
+#include <wolfssl/wolfcrypt/ecc.h>
+#include <wolfssl/wolfcrypt/rsa.h>
+#include <wolfssl/wolfcrypt/asn.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+
+#if (defined(WOLFSSL_MAXQ1065) || defined(WOLFSSL_MAXQ108X)) && \
+    defined(HAVE_PK_CALLBACKS) && defined(WOLF_CRYPTO_CB) && \
+    !defined(NO_PSK) && !defined(HAVE_EXTENDED_MASTER) && \
+    defined(NO_WOLFSSL_SERVER)
+
+/* -------------- */
+/* Configurations */
+/* -------------- */
+
+/* Please define this if you want wolfSSL's debug output */
+#define WANT_DEBUG
+
+/* Please set the server's address and the port it listens on */
+#define DEFAULT_SERVER "127.0.0.1"
+#define DEFAULT_PORT   11111
+
+/* Please define one of USE_ECDHE_ECDSA, USE_FFDHE_RSA, USE_PSK */
+//#define USE_ECDHE_ECDSA
+//#define USE_FFDHE_RSA
+//#define USE_PSK
+
+/* Please define one of USE_TLSV13, USE_TLSV12 */
+//#define USE_TLSV13
+//#define USE_TLSV12
+
+/* Please set the location of the dummy private keys */
```

Comment:
These are public keys. Technically we support using the real public key, but seems like MAXQ is not leveraging that feature and the dummy key is just used to identify the signing algorithm.

Reply:
Yeah, I go into a deeper explanation in the README.md I didn't want to use the real public key because the scripts in the SDK generate lots of artifacts and figuring out which one is the real one is a headache. We spoke about this and you said all wolfSSL needs is the algorithm and parameter set specifier so this is the easiest and simplest way for the end user.